app/controllers/tasks_controller.rb

```diff
@@ -3,17 +3,20 @@
 require 'zip'
 
 class TasksController < ApplicationController # rubocop:disable Metrics/ClassLength
-  load_and_authorize_resource except: %i[import_external import_uuid_check]
+  before_action :load_and_authorize_task, except: %i[index new create import_start import_confirm import_uuid_check import_external]
+  before_action :only_authorize_action, only: %i[import_start import_confirm import_uuid_check import_external]
 
   before_action :handle_search_params, only: :index
   before_action :set_search, only: [:index]
   skip_before_action :verify_authenticity_token, only: %i[import_external import_uuid_check]
+  skip_before_action :require_user!, only: %i[show import_external import_uuid_check]
```

Comment:
If we re-enable the require_user! before action and get an API request without a user we would return an error but not a JSON. I am not sure how we could change that?

Reply:
We already have a JSON handler as part of the `render_error` method. Hence, we just need to ensure to call that. Unfortunately, external users of the API are not necessarily calling a JSON format (i.e., clients such as CodeOcean).

In order to achieve the desired behaviour, we just [define a default in the `routes.rb`](https://guides.rubyonrails.org/routing.html#defining-defaults). When doing so, I correctly got the desired behaviour -- besides the fact that CodeOcean just uses the whole body instead of parsing the JSON file (which I would consider to be part of a follow-up ticket, since it needs to be checked in CodeHarbor and CodeOcean for any of these actions). 

<img width="515" alt="Bildschirmfoto 2024-01-02 um 10 32 50" src="https://github.com/openHPI/codeharbor/assets/7300329/68ca080e-023a-4982-b169-46efd7710f1c">

Reply:
Alright, it was a bit tricky to get the order of before_actions right because I have to set the api user before require_user! is executed. Since the before_action require_user! is defined in the application controller I had to override it. I will push that in a moment.

Reply:
Ah, of course! Besides re-defining a `before_action` with the same name (what you currently do), you can also change the order of the actions and moving `:set_user_for_api_request` to the very first spot. This is done by using `prepend_before_action` rather than `before_action`. [This blog post](https://reganchan.ca/blog/ordering-of-filters-in-rails-controllers/) highlights these differences and the inheritance.

Since prepending will always work for any action to be defined, this is probably the way to go. 

Finally, we can probably drop the comment about "unauthenticated access" in the `TaskPolicy`, can't we?

Reply:
Yes the comment can be dropped. I also switched to prepend_before_action :+1: